modules/tinymce/src/plugins/link/main/ts/ui/Dialog.ts

```diff
@@ -73,10 +142,54 @@ const getInitialData = (info: LinkDialogInfo, defaultTarget: Optional<string>):
     link: url,
     rel: anchor.rel.getOr(''),
     target: anchor.target.or(defaultTarget).getOr(''),
-    linkClass: anchor.linkClass.getOr('')
+    linkClass: anchor.linkClass.getOr(''),
+    fileinput: []
   };
 };
 
+const makeDialogBody = (
+  urlInput: Dialog.UrlInputSpec[],
+  displayText: Dialog.InputSpec[],
+  titleText: Dialog.InputSpec[],
+  catalogs: LinkDialogCatalog,
+  hasUploadPanel: boolean,
+  fileTypes: DocumentsFileTypes[]
+): Dialog.PanelSpec | Dialog.TabPanelSpec => {
+
+  const generalPanelItems = Arr.flatten<Dialog.BodyComponentSpec>([
+    urlInput,
+    displayText,
+    titleText,
+    Optionals.cat([
+      catalogs.anchor.map(ListOptions.createUi('anchor', 'Anchors')),
+      catalogs.rels.map(ListOptions.createUi('rel', 'Rel')),
+      catalogs.targets.map(ListOptions.createUi('target', 'Open link in...')),
+      catalogs.link.map(ListOptions.createUi('link', 'Link list')),
+      catalogs.classes.map(ListOptions.createUi('linkClass', 'Class'))
+    ])
+  ]);
+
+  if (hasUploadPanel) {
```

Comment:
This is adding a feature that anyone can use, not just us. Please create a changelog entry noting that we've added `a new upload tab to the link dialog`, and include this in the JIRA documentation notes as well.

Reply:
done, also created DOC-3327 (already added to the description)